### PR TITLE
refactor(executor): globalize `QueriesPipelineExecutor` creation method

### PR DESCRIPTION
**Pull request by dqhl76**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

**This PR should not affect currently used executor**

After this PR merged:
1. `QueriesPipelineExecutor` becomes a global service.
2. `PipelineExecutor` acts as a client and `QueriesPipelineExecutor` acts as a server. `PipelineExecutor` act a similar role with current using `QueryPipelineExeuctor`

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [x] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

### CODE REVIEW

File: src/query/config/src/config.rs [hunk 1]
```diff
@@ -1656,6 +1656,9 @@ pub struct QueryConfig {
     #[clap(long, value_name = "VALUE", default_value = "0")]
     pub cloud_control_grpc_timeout: u64,
 
+    #[clap(long)]
+    pub enable_queries_executor: bool,
```
Comment (BohuTANG) [inline src/query/config/src/config.rs:1660]: This config is not used.

Comment (dqhl76) [reply src/query/config/src/config.rs:1660]: used here to switch executor: 

https://github.com/datafuselabs/databend/blob/8693b2f95b5beb3baa910d1e556040c43a384fc4/src/query/service/src/global_services.rs#L135-L139

I hard-encoded to test convinently, will use it before PR merged.

Comment (BohuTANG) [reply src/query/config/src/config.rs:1660]: Got it.
BTW, the PR in `draft` state still can do all the benchmark, and it's more safe.
If a PR is `review in ready`, it means all codes ready and can be reviewed :)

Comment (dqhl76) [reply src/query/config/src/config.rs:1660]: Before this PR merged, I will disable the `QueriesPipelineExecutor`. It should not affect currently using executor.

Comment (dqhl76) [reply src/query/config/src/config.rs:1660]: > Got it. BTW, the PR in `draft` state still can do all the benchmark, and it's more safe. If a PR is `review in ready`, it means all codes ready and can be reviewed :)

Thanks for your warm reminding :), will do next time.

**Review by zhang2014 — APPROVED** [review]

**Comment by dqhl76** [comment]
Working on using compile feature to instead of config

**Pull request merged**